packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerConfiguration.ts

```diff
@@ -34,6 +34,19 @@ export function saveLayerRuntimes(layerDirPath: string, runtimes: LayerRuntime[]
   JSONUtilities.writeJson(layerConfigFilePath, layerConfig);
 }
 
+export function getLayerVersionSkip(layerName: string, envName: string): number[] {
+  const layerConfigFilePath = getLayerDirPath(layerName);
+  const layerConfig = JSONUtilities.readJson<$TSAny>(layerConfigFilePath);
+  return _.get<number[]>(layerConfig, [skipVersionsField, envName], []);
+}
+
+export function saveLayerVersionSkip(layerName: string, skipVersions: number[], envName: string) {
+  const layerConfigFilePath = getLayerDirPath(layerName);
+  const layerConfig = JSONUtilities.readJson<$TSAny>(layerConfigFilePath);
+  _.set(layerConfig, [skipVersionsField, envName], skipVersions);
+  JSONUtilities.writeJson(layerConfigFilePath, layerConfig);
+}
+
```

Comment:
Two things:
1) minor nit on naming: I'd go with `getLayerVersionToBeRemovedByCfn` and `saveLayerVersionToBeRemovedByCfn` for clarity.


2) After discussing offline with @attilah, I think we should add an additional top level key that denotes that the section is ephemeral because we want to clear the data once the changes are pushed.

So something like `temporary` or `ephemeral` as a top level key - I'm open to suggestions on the precise naming

```
{
  ...,
  temporary: {
    layerVersionsDelete: {
      <envName>: ...,
    }
  }
}
```